src/NHSD.BuyingCatalogue.Identity.Api/Services/LogoutService.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using IdentityServer4.Events;
+using IdentityServer4.Models;
+using IdentityServer4.Services;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.Services
+{
+    public sealed class LogoutService : ILogoutService
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IIdentityServerInteractionService _identityServerInteractionService;
+        private readonly IEventService _eventService;
+
+        public LogoutService(
+            IHttpContextAccessor httpContextAccessor,
+            IIdentityServerInteractionService identityServerInteractionService,
+            IEventService eventService)
+        {
+            _httpContextAccessor = httpContextAccessor ?? throw new ArgumentNullException(nameof(httpContextAccessor));
+            _identityServerInteractionService = identityServerInteractionService ?? throw new ArgumentNullException(nameof(identityServerInteractionService));
+            _eventService = eventService;
+        }
+
+        public async Task<LogoutRequest> GetLogoutRequest(string logoutId)
+        {
+            return await _identityServerInteractionService.GetLogoutContextAsync(logoutId);
+        }
+
+        public async Task SignOutAsync(LogoutRequest logoutRequest)
+        {
+            HttpContext httpContext = _httpContextAccessor.HttpContext;
+
+            await httpContext.SignOutAsync();
+            await httpContext.SignOutAsync(IdentityConstants.ApplicationScheme);
+
+            httpContext.User = new ClaimsPrincipal(new ClaimsIdentity());
+
+            if (logoutRequest is object)
```

Comment:
Would `logoutRequest != null` not be slightly clearer here? (Purely from intent/readability point-of-view.)

Reply:
I thought this was the new standard for checking not null?

Reply:
Interesting – do you have any references?

Reply:
I had a quick look for a reference but can't find the original (how convenient I know!), did find

https://github.com/dotnet/roslyn/issues/39253

This quote is from the above rabbit hole in answer to "What makes that the best null check in c#? I tend to use "x != null?":

``` 
"That is probably a null check but not guaranteed. If "x" has a custom equality operator then you get what the implementation of that gets. If "x" is a struct then there is a chance it's just evaluated as "false". Basically whether or not it's a null check is subject to context"
```